cpp/src/strings/convert/convert_datetime.cu

```diff
@@ -707,9 +707,9 @@ struct from_timestamp_base {
    *     scale( 61,60) ->  1
    * @endcode
    */
-  __device__ int32_t scale_time(int64_t time, int64_t base) const
+  __device__ int64_t scale_time(int64_t time, int64_t base) const
```

Comment:
This most certainly would have been avoided if this code used `chrono` types directly instead of trying circumvent the types by using integers. 

Filed https://github.com/rapidsai/cudf/issues/9812